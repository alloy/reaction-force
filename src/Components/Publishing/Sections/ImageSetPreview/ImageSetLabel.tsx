import { Flex, Sans, space } from "@artsy/palette"
import React from "react"
import styled from "styled-components"

import { IconImageSet } from "Components/Publishing/Icon/IconImageSet"
import { Media } from "Utils/Responsive"
import { ImageSetPreviewProps } from "./ImageSetPreview"

export const ImageSetLabel = (props: ImageSetPreviewProps) => {
  const {
    section: { images, title },
  } = props
  const label = images.length === 1 ? "Image" : "Images"
  const imageCount = `${images.length} ${label}`
  const primaryTitle = title ? title : imageCount

  return (
    <LabelWrapper alignItems="center" justifyContent="space-between">
      <Flex flexDirection="column" justifyContent="space-between">
        <Sans size={["4", "5"]} weight="medium" pb={2}>
          {primaryTitle}
        </Sans>

        <Flex>
          <Sans size={["2", "3"]} weight="medium">
            View Slideshow
          </Sans>
          {title && (
            <Sans size={["2", "3"]} pl={20}>
              {imageCount}
            </Sans>
          )}
        </Flex>
      </Flex>

      <Media greaterThanOrEqual="sm">
        <IconContainer>
          <IconImageSet />
        </IconContainer>
      </Media>
    </LabelWrapper>
  )
}

export const IconContainer = styled.div`
  height: 45px;
  position: relative;
  margin-left: ${space(4)}px;
  text-align: right;

  > svg {
    height: 98%;
  }
`

export const LabelWrapper = styled(Flex)`
  padding: ${space(2)}px;
  width: 100%;
`
