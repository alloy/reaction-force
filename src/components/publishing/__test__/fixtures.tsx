export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini = {
  type: "image_set",
  layout: "mini",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const FeatureHeaders = [
  {
    layout: "text",
    title: "Looking Into the Future",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    layout: "split",
    title: "New York's Next Art District",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    layout: "full",
    title: "New York's Next Art District",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    vertical: "Market Analysis",
    author: "Daniel Kunitz",
    date: "Jan 4, 2017",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]
