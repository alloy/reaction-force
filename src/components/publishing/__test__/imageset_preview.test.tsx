import ImageSetPreview from "../imageset_preview"
import { ImageSetFull, ImageSetMini } from "./fixtures"

import * as React from "react"
import * as renderer from "react-test-renderer"

it("renders a full image set properly", () => {
  const imageset = renderer.create(<ImageSetPreview section={ImageSetFull} />).toJSON()
  expect(imageset).toMatchSnapshot()
})
it("renders a mini image set properly", () => {
  const imageset = renderer.create(<ImageSetPreview section={ImageSetMini} />).toJSON()
  expect(imageset).toMatchSnapshot()
})
