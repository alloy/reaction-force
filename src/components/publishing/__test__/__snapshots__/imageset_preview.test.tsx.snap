// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a full image set properly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="file__FullWrapper-s1x4ygsx-0 jcuYwD"
  >
    <div
      className="file__TitleWrapper-s1x4ygsx-3 kREUlU"
      data-layout="full"
    >
      <div
        className="file__Title-s1x4ygsx-4 erllsr"
      >
        The Work of Bruce M. Sherman
      </div>
      <div
        className="file__SubTitle-s1x4ygsx-5 dlPgRc"
      >
        <div
          className="file__SubTitlePrompt-s1x4ygsx-6 fHTrtu"
        >
          View Slideshow
        </div>
        <div
          className="file__SubTitleCount-s1x4ygsx-7 fwNFCM"
        >
          3
           Images
        </div>
      </div>
    </div>
    <div
      className="file__IconContainer-s1x4ygsx-8 cLLfEt"
    >
      <svg
        data-name="imageset"
        id="imageset"
        viewBox="0 0 44 44"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
          transform="translate(-6728.67 -507.72)"
        />
        <polygon
          points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
        />
        <polygon
          points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
        />
      </svg>
    </div>
  </div>
  <img
    src="https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg"
    width="100%"
  />
</div>
`;

exports[`renders a mini image set properly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="file__MiniWrapper-s1x4ygsx-1 YMOHT"
  >
    <img
      height="100%"
      src="https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg"
      width="auto"
    />
    <div
      className="file__MiniInner-s1x4ygsx-2 dzwRNi"
    >
      <div
        className="file__TitleWrapper-s1x4ygsx-3 kREUlU"
        data-layout="mini"
      >
        <div
          className="file__Title-s1x4ygsx-4 erllsr"
        >
          The Work of Bruce M. Sherman
        </div>
        <div
          className="file__SubTitle-s1x4ygsx-5 dlPgRc"
        >
          <div
            className="file__SubTitlePrompt-s1x4ygsx-6 fHTrtu"
          >
            View Slideshow
          </div>
          <div
            className="file__SubTitleCount-s1x4ygsx-7 fwNFCM"
          >
            3
             Images
          </div>
        </div>
      </div>
      <div
        className="file__IconContainer-s1x4ygsx-8 cLLfEt"
      >
        <svg
          data-name="imageset"
          id="imageset"
          viewBox="0 0 44 44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
            transform="translate(-6728.67 -507.72)"
          />
          <polygon
            points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
          />
          <polygon
            points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
